### PR TITLE
Migration fixes

### PR DESCRIPTION
**Pull request by beylmk**
Leaving this in draft due to a number of questions, but could use some eyes...

* Adds back deprecation warnings -> these are causing build errors in objc tester, see comments
* Renames `completionBlock` argument labels -> `completion`...do we want to rename for objc too?
* Restores "omittability" of external argument labels from pre-migration
* A few updated "maybe" prefixes on nullable parameters

### CODE REVIEW

File: Purchases/Public/Purchases.swift
Comment (beylmk) [inline Purchases/Public/Purchases.swift:178]: All of these new deprecations are causing build errors in the ObjC API tester. I'm a little confused why this is happening, since `debugLogsEnabled` is called, and builds with just an error. Perhaps there's a difference in treatment between properties and functions? Or did we ever configure something to treat certain deprecation errors as warnings?

Either way, I'd think warnings are preferable unless we want to officially drop support for those methods.

I confirmed that in Swift, these only trigger build warnings...

File: Purchases/Public/Purchases.swift
Comment (beylmk) [inline Purchases/Public/Purchases.swift:621]: is this 3rd version of the function still necessary? were we avoiding breaking changes with the argument labels?

Comment (taquitos) [reply Purchases/Public/Purchases.swift:621]: I was avoiding breaking changes, but really... maybe we should remove this? We are shipping a new major version.

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:178]: #pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
    // deprecate debugLogsEnabled call here
#pragma clang diagnostic pop

Comment (taquitos) [reply Purchases/Public/Purchases.swift:178]: I turned on build warnings as errors in the API tester project. Context:
If we change the nullability of something and we pass in `nil` in objc, it's not a build error- just a warning.
(https://github.com/RevenueCat/purchases-ios/pull/827)

Comment (beylmk) [reply Purchases/Public/Purchases.swift:178]: > ```
> #pragma clang diagnostic push
> #pragma clang diagnostic ignored "-Wdeprecated-declarations"
>     // deprecate debugLogsEnabled call here
> #pragma clang diagnostic pop
> ```

we don't have that around the current debugLogs call, though, which is what was confusing me...although now this doesn't appear to be triggering a warning either 🤯 :
`debugLogsEnabled = [RCPurchases debugLogsEnabled];`

File: Purchases/Public/Purchases.swift
Comment (beylmk) [reply Purchases/Public/Purchases.swift:621]: either that, or align the swift parameter labels with those from before and add an @objc version of the method name...kinda what we did with optional external parameter namings, or else one of the APIs had to break.

File: Purchases/Public/Purchases.swift
Comment (beylmk) [reply Purchases/Public/Purchases.swift:178]: welp, not sure what could've changed but now i'm getting build _warnings_ for the ones i added, and still nothing for `debugLogsEnabled`.

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift]: I just took a look at this again, we can't get rid of it yet, we don't have a `setAppleSearchAdsData` method so we can't actually redirect folks to the `set<network>` methods

**Review by taquitos — APPROVED** [review]
I think this is ready to go. We can talk about renaming the ObjC block parameters and do another PR if we decide to change that. I kinda think we should since that rename should be super easy for ObjC using folks.

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:178]: Yeah, that makes sense. I think for the tester project that's ok. It would be nice to be able to specify places where we want to emit a warning or an error.

**Review by taquitos — APPROVED** [review]
🐐 (= ФェФ=)

File: APITesters/ObjCAPITester/ObjCAPITester/RCBackendErrorCodeAPI.m
Comment (aboedo) [inline APITesters/ObjCAPITester/ObjCAPITester/RCBackendErrorCodeAPI.m:20]: kinda surprised that these are public in the first place, I think it might have been an oversight on our part

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (aboedo) [inline APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift:20]: question about our API testers: 
both of them are using abbreviations for variable names. Is there a particular reason for this? We never do abbreviations for any other part of the code (if anything, we have a tendency to go overly verbose at times).

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (aboedo) [inline APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift:45]: `type` has a special meaning in Swift, perhaps periodType is better in this context

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (aboedo) [inline APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift:74]: any reason these are commented out?

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (aboedo) [inline APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift:163]: I see, this is the kind of thing why we wanted to separate the api testers into a separate project, right? 
otherwise, folks using Xcode direct integration would have to deal with these

File: PublicSDKAPITester/RCPurchasesErrorCodeAPI.m
Comment (aboedo) [inline PublicSDKAPITester/RCPurchasesErrorCodeAPI.m:22]: I think it'd make sense to also check the values for these, we've had issues in the past where we messed up the values for errors and that translates into a bad mapping for the hybrids

File: PublicSDKAPITester/RCPurchasesErrorUtilsAPI.m
Comment (aboedo) [inline PublicSDKAPITester/RCPurchasesErrorUtilsAPI.m:34]: Is this intentional?

File: PurchasesTests/Mocks/MockSKProduct.swift
Comment (aboedo) [inline PurchasesTests/Mocks/MockSKProduct.swift:37]: note for other reviewers: the `#available` check wasn't needed, since `SKProductDiscount` is available on 11.2+

File: Purchases/Public/Purchases.swift
Comment (aboedo) [inline Purchases/Public/Purchases.swift:29]: since the typealiases won't be available for obj-c, perhaps we can just remove them? are we getting value out of them?

File: Purchases/Public/Purchases.swift
Comment (aboedo) [inline Purchases/Public/Purchases.swift:649]: although I love the `maybe` convention, I'm not sure we should use it for `public` methods, since it's not standard. 

Also, I think we should have the same names for objc and swift, since it'd be confusing for folks who might be migrating their obj-c classes to swift, when they find that our method naming isn't consistent

**Review by aboedo — COMMENTED** [review]
I left a few comments. I don't want to hold this PR on them, but I really think we should take this opportunity to standardize naming for all our APIs and make it consistent across the board. 
We won't get to update our existing API often since we'll always want to minimize breaking changes, so I think we need to make the most out of this major

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (taquitos) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift:45]: 💯

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:29]: That's a great point, internally, I think they are valuable. I can't remember exactly which at the moment, but one of them wraps another and removing them altogether make it look like a confusing completionblock-ception.

Comment (aboedo) [reply Purchases/Public/Purchases.swift:29]: makes sense. Maybe we can remove the simpler ones, and make the others internal, if we're not sure they'll add value to the end user?

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:649]: Yeah, that's a good point. For public methods, we should probably go back to just `completion`

File: PublicSDKAPITester/RCPurchasesErrorUtilsAPI.m
Comment (taquitos) [reply PublicSDKAPITester/RCPurchasesErrorUtilsAPI.m:34]: I do not think so.

File: PublicSDKAPITester/RCPurchasesErrorCodeAPI.m
Comment (taquitos) [reply PublicSDKAPITester/RCPurchasesErrorCodeAPI.m:22]: Makes sense.

File: Purchases/Public/Purchases.swift
Comment (vegaro) [inline Purchases/Public/Purchases.swift:948]: The label is `maybeCompletion` in `syncPurchases`

File: Purchases/Public/Purchases.swift
Comment (vegaro) [inline Purchases/Public/Purchases.swift:972]: This completion block's naming doesn't match the others where we do:

```
  _ maybeCompletion: @escaping ReceiveIntroEligibilityBlock) {
```

File: Purchases/Public/Purchases.swift
Comment (vegaro) [inline Purchases/Public/Purchases.swift:1097]: I believe this would be `maybeCompletion`

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (taquitos) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift:20]: I started doing that as shorthand because the api tester variables were never meant to do anything except provide the compiler something. I think short vars when it comes to these tools is ok because we aren't actually using them. If we were to make these projects runnable, or contain any logic, then I would advocate for changing to names to things that make sense. Thoughts?

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:1099]: Updated all `public` completion blocks to using just `completion` instead of `maybeCompletion`

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:950]: Updated param `maybeCompletion` in favor of `completion`

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift]: Updated all `completionBlock` to `completion` and removed `maybe` for all public api, as well.

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:29]: Double checked these, removed most, left only two public ones because they are actually used for our API and helpful IMO.

File: PublicSDKAPITester/RCPurchasesErrorUtilsAPI.m
Comment (taquitos) [reply PublicSDKAPITester/RCPurchasesErrorUtilsAPI.m]: ✅

File: APITesters/ObjCAPITester/ObjCAPITester/RCBackendErrorCodeAPI.m
Comment (taquitos) [reply APITesters/ObjCAPITester/ObjCAPITester/RCBackendErrorCodeAPI.m:20]: We can remove them, but they have been public for a while, so I'm apprehensive about it. Regardless, I think that's another PR if we decide.

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift]: ✅

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift]: I'm actually going a different direction with this one, shortening it to just `completion`

File: PublicSDKAPITester/RCPurchasesErrorCodeAPI.m
Comment (taquitos) [reply PublicSDKAPITester/RCPurchasesErrorCodeAPI.m]: I think we'll need to add that to unit tests.

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (taquitos) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift]: Yeah, I believe they might end up needing to deal with the warnings.

Comment (aboedo) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift]: they wouldn't, though, if we move our API testers into a separate project, right? 
we can have them live in a separate project and then create an xcworkspace to incorporate them all, so that we can still easily test stuff, while making it easy for other folks to integrate

Comment (taquitos) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift]: So, one would think this is right, but it appears that if we use SPM in one project for a dependency in the workspace, and not in another, then Xcode doesn't like that. Right now the APITesters use SPM for their dependency. I don't remember the motivation behind that, but if we were to just rely on the RevenueCat.xcodeproj in a workspace instead, then 100% you're right.

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (taquitos) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift]: There doesn't seem to be. Updated.

File: PublicSDKAPITester/RCPurchasesErrorCodeAPI.m
Comment (taquitos) [reply PublicSDKAPITester/RCPurchasesErrorCodeAPI.m]: Added unit tests for this in RevenueCat project (ErrorCodeTests.swift)

File: APITesters/APITesters.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (aboedo) [inline APITesters/APITesters.xcworkspace/xcshareddata/swiftpm/Package.resolved:16]: is this one supposed to get updated on every commit? if so, perhaps we should not version it or add automation?

Comment (aboedo) [reply APITesters/APITesters.xcworkspace/xcshareddata/swiftpm/Package.resolved:16]: why does it use SPM for the dependency?

File: APITesters/SwiftAPITester/SwiftAPITester.xcodeproj/project.pbxproj
Comment (aboedo) [inline APITesters/SwiftAPITester/SwiftAPITester.xcodeproj/project.pbxproj:355]: same thoughts regarding SPM here, how does this flow work once it's merged? do we manually update the branch?

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (aboedo) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift:20]: I'm not a fan of abbreviations in any context tbh, they just look odd to me and are harder to read than the actual word. Particularly for lines like [this one](https://github.com/RevenueCat/purchases-ios/pull/839/files#diff-f0a38d89d6f32bb0701941d36f39aeb58b93e3363e6d387e4c805a5f034c4190R24). 
I also feel a bit apprehensive of treating any piece of our code as a second-class citizen, since folks might be tempted to copy / paste or look at our API tester as a reference for API usage
Not a big deal, though.

File: Purchases/Public/Purchases.swift
Comment (aboedo) [inline Purchases/Public/Purchases.swift:622]: I think if we use the `renamed` attribute in `@available`, xcode auto-suggests the fix-it, which would be cool

File: Purchases/Public/Purchases.swift
Comment (aboedo) [inline Purchases/Public/Purchases.swift:727]: not sure the `_`s are a good idea - folks can already skip the completion block by just doing 
```swift
reset { purchaserInfo, error in
    // 
}
```

So I think it'd be fine to keep the block names, and if we did, we'd also get rid of the `@objc` names... which means that we're providing the exact same interface for obj-c and swift, which is a win I think

File: Purchases/Public/RevenueCat.h
Comment (aboedo) [inline Purchases/Public/RevenueCat.h:26]: should we just get rid of this file? Is it needed?

File: Purchases/Public/Purchases.swift
Comment (aboedo) [inline Purchases/Public/Purchases.swift:1071]: absolutely love the consistency in these now

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:727]: 💯

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:622]: 💯

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift]: I just made a few updates for this across our public API 😄

File: Purchases/Public/RevenueCat.h
Comment (taquitos) [reply Purchases/Public/RevenueCat.h]: 💯, deleted.

File: Purchases/Public/Purchases.swift
Comment (taquitos) [reply Purchases/Public/Purchases.swift:1065]: Yeah, I'm a fan 😃

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (taquitos) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift]: https://github.com/RevenueCat/purchases-ios/issues/857

File: APITesters/APITesters.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (taquitos) [reply APITesters/APITesters.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Will follow up with https://github.com/RevenueCat/purchases-ios/issues/857

File: APITesters/SwiftAPITester/SwiftAPITester.xcodeproj/project.pbxproj
Comment (taquitos) [reply APITesters/SwiftAPITester/SwiftAPITester.xcodeproj/project.pbxproj:355]: Will follow up with https://github.com/RevenueCat/purchases-ios/issues/857

**Review by taquitos — COMMENTED** [review]
Merging 🎉

File: APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift
Comment (beylmk) [reply APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift]: As of now they just use SPM because it's the first integration I tried 🤷  I was considering trying to create testers using carthage and cocoapods as well, at least for testing the migration guide locally. That's where @taquitos was helping me figure out how to have a shared folder between different xcode projects, so hopefully only the integration piece needs to be separate.

Longer term, I see the API testers as purely testing our API...there wouldn't be any differences in our API between integrations, right?

File: APITesters/SwiftAPITester/SwiftAPITester.xcodeproj/project.pbxproj
Comment (beylmk) [reply APITesters/SwiftAPITester/SwiftAPITester.xcodeproj/project.pbxproj:355]: yeah, had it on my list for the commit to be updated to the latest PR commit in the circleci job. i'll make a ticket for that.

for now, i wanted to leave it pointing to this particular commit for script testing.

**Pull request merged**
